api/src/test/java/org/openmrs/util/databasechange/H2LessStrictDialect.java

```diff
@@ -22,20 +21,40 @@
 	public H2LessStrictDialect() {
 		super();
 		
-		//H2Dialect incorrectly sets these to synonyms in H2
-		registerColumnType(Types.LONGVARCHAR, "varchar");
+		// H2Dialect incorrectly sets these to synonyms in H2
+		//
 		registerColumnType(Types.BIGINT, "integer");
 		
-		//Liquibase incorrectly creates varchar for clob in H2 so we just tell Hibernate it's ok
+		// Liquibase incorrectly creates varchar for clob in H2 so we just tell Hibernate it's ok
+		//
 		registerColumnType(Types.CLOB, "varchar");
 		
-		//Our UUIDs are created as char(38), but H2Dialect maps them to varchars
-		registerColumnType(Types.VARCHAR, 38, "char($1)");
-		
-		//Float is mapped to real per H2 docs at http://www.h2database.com/html/datatypes.html#real_type
-		registerColumnType(Types.FLOAT, "real");
+		// H2 maps 'FLOAT' to 'double' as per http://www.h2database.com/html/datatypes.html#double_type
+		//
+		// Without mapping 'float' to 'double' the validation of Hibernate mappings fails:
+		//   Schema-validation: 
+		//     wrong column type encountered in column [sort_weight] in table [form_field]; 
+		//     found [double (Types#DOUBLE)], but expecting [float (Types#FLOAT)]
+		//
+		registerColumnType(Types.FLOAT, "double");
 		
 		//person.birthdate is not a timestamp, but date in db
+		//
 		registerColumnType(Types.TIMESTAMP, "date");
+		
+		// UUIDs are created as char(38), but H2Dialect maps them to varchars
+		//
+		registerColumnType(Types.VARCHAR, 38, "char($1)");
+		
+		// These mappings are required for "long" fields of type java.lang.String that are declared as 'text' 
+		// in Hibernate change sets.
+		//
+		registerColumnType(Types.VARCHAR, 250, "clob");
```

Comment:
What led to these column type registrations? The liquibase upgrade?

Reply:
Yes, the Liquibase upgrade resulted in H2 and Hibernate mappings behaving differently than before.